Tests/Unit/PushStateTests.m

```diff
@@ -120,9 +123,15 @@ - (void)testSetPayloadWithMessage {
     PFMutablePushState *state = [[PFMutablePushState alloc] init];
     [state setPayloadWithMessage:@"yolo"];
     XCTAssertEqualObjects(state.payload, @{ @"alert" : @"yolo" });
-
+    
     [state setPayloadWithMessage:nil];
     XCTAssertNil(state.payload);
 }
 
+- (void)testSetPushTimeValidation {
+    PFMutablePushState *state = [[PFMutablePushState alloc] init];
+    PFAssertThrowsInvalidArgumentException(state.pushDate = [NSDate dateWithTimeIntervalSinceNow:-10]);
```

Comment:
These are the kind of tests that could end up being flaky on travis if we get a stupidly slow VM. 

I'd strongly prefer using a concrete date in the past (maybe the unix epoch) for the first test, and `distantFuture` for the second test.